jetty-util/src/main/java/org/eclipse/jetty/util/thread/strategy/AdaptiveExecutionStrategy.java

```diff
@@ -134,20 +135,27 @@ public class AdaptiveExecutionStrategy extends ContainerLifeCycle implements Exe
     private final LongAdder _epcMode = new LongAdder();
     private final Producer _producer;
     private final Executor _executor;
+    private final boolean _useVirtualThreads;
     private final TryExecutor _tryExecutor;
     private final Runnable _runPendingProducer = () -> tryProduce(true);
     private State _state = State.IDLE;
     private boolean _pending;
 
     /**
-     * @param producer The produce of tasks to be consumed.
+     * @param producer The producer of tasks to be consumed.
      * @param executor The executor to be used for executing producers or consumers, depending on the sub-strategy.
      */
     public AdaptiveExecutionStrategy(Producer producer, Executor executor)
+    {
+        this(producer, executor, false);
+    }
+
+    public AdaptiveExecutionStrategy(Producer producer, Executor executor, boolean useVirtualThreads)
     {
         _producer = producer;
         _executor = executor;
-        _tryExecutor = TryExecutor.asTryExecutor(executor);
+        _useVirtualThreads = useVirtualThreads;
+        _tryExecutor = useVirtualThreads ? TryExecutor.NO_TRY : TryExecutor.asTryExecutor(executor);
```

Comment:
yeah... but no!

If you don't want to use reserved threads, then configure 0 reserved threads.   If reserved threads are configured in the thread pool, then they should be used if available.

No need to do anything special here as the user already can configure a NO_TRY tryExecutor if they want it.